.github/workflows/azure.yml

```diff
@@ -11,8 +11,54 @@ defaults:
     shell: bash
 
 jobs:
-  sgx-hw:
+  aks:
     runs-on: self-hosted
+    env:
+      CLUSTER_NAME_BASE: gha-cluster
+    strategy:
+      fail-fast: true
+      # Running two Kubernetes clusters side-by-side from the same user in the
+      # same machine creates conflicts in the local .kubeconfig file. A
+      # possible fix would be to use one .kubeconfig file per cluster, and make
+      # all k8s commands point to that .kubeconfig file. Given that this
+      # workflow is not latency-sensitive, we actually don't mind that jobs are
+      # run one after the other, so we don't implement the fix.
+      max-parallel: 1
+      matrix:
+        sgx: [True, False]
     steps:
-      - name: "Checck out the experiment-base code"
+      - name: "Check out the experiment-base code"
         uses: actions/checkout@v3
+        with:
+          repository: faasm/experiment-base
+          path: experiment-base
+      - name: "Create a unique cluster name"
+        run: |
+          [[ "${{ matrix.sgx }}" == "true" ]] && SUFFIX_SGX="-sgx" || SUFFIX_SGX=""
+          echo "CLUSTER_NAME=${CLUSTER_NAME_BASE}${SUFFIX_SGX}-${{ github.job }}-${{ github.run_id }}-${{ github.run_attempt }}" >> $GITHUB_ENV
+      - name: "Install kubectl"
+        run: ./bin/inv_wrapper.sh k8s.install-kubectl
+        working-directory: ${{ github.workspace }}/experiment-base
+      - name: "Start a managed k8s cluster on Azure's Kubernetes Service"
+        run: |
+          ./bin/inv_wrapper.sh cluster.provision --name ${{ env.CLUSTER_NAME }} --vm Standard_DC4s_v3 --nodes 4 --location eastus2 --sgx ${{ matrix.sgx }}
+          ./bin/inv_wrapper.sh cluster.credentials --name ${{ env.CLUSTER_NAME }}
+        working-directory: ${{ github.workspace }}/experiment-base
+      - name: "Check out faasm code"
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+          path: faasm
+      - name: "Deploy Faasm on k8s cluster"
+        run: ./bin/inv_wrapper.sh k8s.deploy --workers=4 --sgx ${{ matrix.sgx }}
+        working-directory: ${{ github.workspace }}/faasm
+      - name: "Build, upload and run a simple CPP function"
+        run: docker compose -f docker-compose-k8s.yml run -T cpp-cli ./bin/inv_wrapper.sh func demo hello func.upload demo hello func.invoke demo hello
+        working-directory: ${{ github.workspace }}/faasm
+      - name: "Delete cluster"
+        if: always()
+        run: ./bin/inv_wrapper.sh cluster.delete --name ${{ env.CLUSTER_NAME }}
+        working-directory: ${{ github.workspace }}/experiment-base
+      - name: "Chown all files"
```

Comment:
This is necessary to ensure that consecutive runs on the same runner can clean the state (i.e. remove all files). Otherwise, some docker-generated files may be `root`-owned.